### PR TITLE
Cherry pick PRs for 2.3.1

### PR DESCRIPTION
**Pull request by rreusser**
Merges the following PRs for release 2.3.1:

https://github.com/mapbox/mapbox-gl-js/pull/10779
https://github.com/mapbox/mapbox-gl-js/pull/10772
https://github.com/mapbox/mapbox-gl-js/pull/10767

### CODE REVIEW

File: src/style-spec/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 13.20.1
+
+### 🐞 Bug fixes
+* Increase strictness of the style API validation ([#10779](https://github.com/mapbox/mapbox-gl-js/pull/10779))
```
Comment (karimnaaji) [inline src/style-spec/CHANGELOG.md:4]:
```suggestion
* Increase strictness of the style API validation for source types ([#10779](https://github.com/mapbox/mapbox-gl-js/pull/10779))
```

File: src/style-spec/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+## 13.20.1
+
+### 🐞 Bug fixes
```
Comment (karimnaaji) [inline src/style-spec/CHANGELOG.md:3]:
```suggestion
### 🐞 Bug fixes

```
Add extra line

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
+## 2.3.1
+
+### 🐞 Bug fixes
+* Increase strictness of the style API validation ([#10779](https://github.com/mapbox/mapbox-gl-js/pull/10779))
+* Remove strictly-increasing requirement for fog range validation ([#10772](https://github.com/mapbox/mapbox-gl-js/pull/10772))
+* Fix fog flickering when `optimizeForTerrain` is false ([#10763](https://github.com/mapbox/mapbox-gl-js/pull/10767))
```
Comment (karimnaaji) [inline CHANGELOG.md:6]: Small suggestion:
```suggestion
* Fix fog flickering when the map option `optimizeForTerrain` is set to false ([#10763](https://github.com/mapbox/mapbox-gl-js/pull/10767))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,11 @@
+## 2.3.1
+
+### 🐞 Bug fixes
+* Increase strictness of the style API validation ([#10779](https://github.com/mapbox/mapbox-gl-js/pull/10779))
+* Remove strictly-increasing requirement for fog range validation ([#10772](https://github.com/mapbox/mapbox-gl-js/pull/10772))
```
Comment (karimnaaji) [inline CHANGELOG.md:5]: I think we can remove the spec-only changes as they are in the other changelog. The second one is more debatable as it touches a little bit of code in the fog culling, but seems mainly spec focused overall.
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,11 @@
+## 2.3.1
+
+### 🐞 Bug fixes
```
Comment (karimnaaji) [inline CHANGELOG.md:3]: Add extra line:
```suggestion
### 🐞 Bug fixes

```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,11 @@
...
+* Increase strictness of the style API validation ([#10779](https://github.com/mapbox/mapbox-gl-js/pull/10779))
+* Remove strictly-increasing requirement for fog range validation ([#10772](https://github.com/mapbox/mapbox-gl-js/pull/10772))
+* Fix fog flickering when `optimizeForTerrain` is false ([#10763](https://github.com/mapbox/mapbox-gl-js/pull/10767))
+
+
```
Comment (karimnaaji) [inline CHANGELOG.md:8]: Remove extra line:
```suggestion
```

File: src/style-spec/CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,10 @@
...
+* Increase strictness of the style API validation ([#10779](https://github.com/mapbox/mapbox-gl-js/pull/10779))
+* Remove strictly-increasing requirement for fog range validation ([#10772](https://github.com/mapbox/mapbox-gl-js/pull/10772))
+
+
```
Comment (karimnaaji) [inline src/style-spec/CHANGELOG.md:7]: Remove extra line:
```suggestion
```

File: CHANGELOG.md (hunk 4 above)
Comment (rreusser) [reply CHANGELOG.md:5]: 👍 Yeah, I wondered about that. Sounds good.

**Review by karimnaaji — APPROVED** [review]

**Pull request merged**